### ISSUE TITLE
Style/MultilineMemoization autocorrects to invalid ruby

### ISSUE THREAD

**Issue by nikolai-b** [bbatsov/rubocop#4322]
## Expected behavior

Style/MultilineMemoization auto-corrects when the `(` are not on new lines by replacing them with `begin` 

## Actual behavior

If `(` is not on its own line then a new line should be added.  

## Steps to reproduce the problem

Autocorreting
```ruby
class MyClass
  def my_method
    @my_method ||= (true ||
                    true ||
                    false)
  end
end
```
becomes
```ruby
class MyClass
  def my_method
    @my_method ||= begintrue ||
                    true ||
                    falseend
  end
end
```

## RuboCop version

```
$ rubocop -V
0.43.0 (using Parser 2.4.0.0, running on ruby 2.2.4 x86_64-linux)
```

**Comment by ilansh** [comment]
I'm looking into fixing this.
Assuming the expected result would be (correct me if I'm wrong):
```ruby
class MyClass
  def my_method
    @my_method ||= begin
      true ||
      true ||
      false
    end
  end
end
```
Should the cop also fix the indentation of items in the expression, or is it ok to let other cops fix it?

**Comment by Drenmi** [comment]
@ilansh: Basic indentation is generally handled by each individual cop's autocorrect. 🙂

**Comment by nikolai-b** [comment]
@ilansh I've got a PR open in #4323 but I wasn't sure how best to do the indentation.
